discretisedfield/plotting/mpl_field.py

```diff
@@ -872,9 +872,87 @@ def __dir__(self):
 
         return dirlist
 
-    def _add_colorbar(self, ax, cp, colorbar_label):
+    def _add_colorbar(
+        self,
+        ax,
+        cp,
+        colorbar_label,
+        min_height_inches=2.0,
+        min_width_inches=0.5,
+        min_pad_inches=0.5,
+    ):
+        """Adds a colorbar to the current plot.
+
+        This colorbar has a minimum width, height, and padding. Multiple colorbars
+        can be added. These additional colorbars will have the same width and height
+        as the original colorbar.
+        """
+
+        # Need to normalise physical units
+        fig = ax.figure
+        fig_width_inches, fig_height_inches = fig.get_size_inches()
+
+        pos = ax.get_position()
+
+        # relative to ax
+        min_height_norm = min_height_inches / (fig_height_inches * (pos.y1 - pos.y0))
+        # relative to fig
+        min_width_norm = min_width_inches / fig_width_inches
+        min_pad_norm = min_pad_inches / fig_width_inches
+
+        # Calculate pad and width
+        if min_pad_norm > 0.05:
+            pad_h = Size.Fixed(min_pad_inches)
+        else:
+            pad_h = Size.AxesX(ax, aspect=0.05)
+
+        if min_width_norm > 0.05:
+            width_h = Size.Fixed(min_width_inches)
+        else:
+            width_h = Size.AxesX(ax, aspect=0.05)
+
+        if min_height_norm > 1:
+            v_aspect = min_height_norm
+        else:
+            v_aspect = 1
```

Comment:
I would find some additional comments explaining the ideas behind the individual parts useful.